presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainOrcPredicate.java

```diff
@@ -75,9 +96,149 @@ public boolean matches(long numberOfRows, Map<Integer, ColumnStatistics> statist
+        if (effectivePredicateDomains.isEmpty()) {
+            // no predicate, probably a where-less query, read
+            return true;
+        }
+        Map<C, Domain> stripDomains = optionalStripeDomains.get();
+
+        int numberOfBloomFiltersChecked = 0;
+        for (ColumnReference<C> columnReference : columnReferences) {
+            // is this part of a predicate?
+            if (!effectivePredicateDomains.containsKey(columnReference.getColumn())) {
+                continue;
+            }
+            if (!stripDomains.containsKey(columnReference.getColumn())) {
+                continue;
+            }
+
+            ColumnStatistics columnStatistics = statisticsByColumnIndex.get(columnReference.getOrdinal());
+            if (columnStatistics == null) {
+                // need column statistics, treat as failure: read
+                return true;
+            }
+
+            List<RowGroupBloomfilter> bloomfilters = columnStatistics.getBloomfilters();
+            if (bloomfilters == null || bloomfilters.isEmpty()) {
+                // need bloom filters, treat as failure: read
+                return true;
+            }
+
+            Domain domain = effectivePredicateDomains.get(columnReference.getColumn());
+            if (domain.isAll()) {
+                // all values are okay, go to next one as we don't have to check the bf
+                continue;
+            }
+
+            // extract values from domain
+            Collection<Object> predicateValues = predicateValuesFromDomain(domain);
+            if (predicateValues == null || predicateValues.isEmpty()) {
+                // no values checked, treat as failure: read
+                return true;
+            }
+
+            // run values against the bloomfilters: none can match in order to skip a read
+            for (Object predicateValue : predicateValues) {
+                for (RowGroupBloomfilter rowGroupBloomfilter : bloomfilters) {
+                    BloomFilter bloomfilter = rowGroupBloomfilter.getBloomfilter();
+
+                    boolean matched = checkInBloomFilter(bloomfilter, predicateValue);
+                    numberOfBloomFiltersChecked++;
+                    if (matched) {
+                        // bloom filter is matched here return true so we select this stripe as it likely contains data which we need to read
+                        return true;
+                    }
+                }
+            }
+        }
+
+        // did we read any bloom filter?
+        if (numberOfBloomFiltersChecked < 1) {
+            // no bloom filters read, seems like a query that was run only on partitioned columns
+            return true;
+        }
+
+        // none of the bloomfilters caused a "hit" meaning we should not read
+        log.debug("Not reading, didn't match any of the bloomfilters (" + numberOfBloomFiltersChecked + "x), data is not here");
```

Comment:
Remove this. We don't log things like this, even at debug level, since they'll flood the logs (a single query with 100k splits would produce 100k log lines).